VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTopic.java

```diff
@@ -80,7 +83,9 @@ public boolean hasSameId(@NonNull DiscussionTopic discussionTopic) {
     }
 
     public boolean containsThread(@NonNull DiscussionThread discussionThread) {
-        if (identifier.equals(discussionThread.getTopicId())) {
+        if (isAllType() || isFollowingType()) {
```

Comment:
> we can't be on the "Posts I'm following" and "All Posts" screens at once for the condition to be true.

That's not what I was suggesting, think you may have misread. I'm only suggesting that, if the topic `isFollowingType()` is true, then `containtsThread()` should only return true if the `discussionThread` passed is being followed (`isFollowing() == true`)